### ISSUE TITLE
File doesn't save in Facebook App browser

### ISSUE THREAD

**Issue by p3lican** [eligrey/FileSaver.js#649]
Works in Chrome but file doesn't save in the stupid Facebook browser.

Anyone know why other than that the FB browser sux? It's my biggest audience though. 😓

**Comment by jimmywarting** [comment]
Facebook browser? you mean some stupid webview on android/iOS?

**Comment by p3lican** [comment]
Yeah exactly. If I type the url into a post on Facebook and someone clicks on it, it defaults to webview with no address bar, just a back/exit button. File download doesn't trigger there.

Most people don't know you can easily click the top-right ellipsis in FB and "open in chrome" or change the default FB link opening setting.

Just wondering if anyone had a similar issue and was able to solve it with a workaround.

**Comment by islam-mohammed** [comment]
@p3lican did you find any solution for this issue? I am having the same !

**Comment by mgoadric** [comment]
I am also seeing the same issue. I'm using FileSaver in combination with dom-to-image, and while the link works perfectly in Chrome on my Android device, but won't work in the Facebook webview. I'm not even sure how to debug it.

**Comment by biasto** [comment]
I have the same issue, has anybody been able to find a workaround?

**Comment by usb248** [comment]
Same bug.... Very disturbing @eligrey  any workaround to fix this behaviour ?

**Comment by usb248** [comment]
For some people who search a solution :
https://github.com/f2etw/detect-inapp/blob/master/src/inapp.js
this package can help you to detect if your user is in messenger/facebook/insta etc... and change the UI according to this.

**Comment by p3lican** [comment]
Sorry, I never did find a work-around.
Would still love to use this functionality but a little scared it'll flop with my FB users.
I don't know how this works under the hood, I'm just appreciative that it does work. It's super cool.
Would this apply here in any way? [https://stackoverflow.com/a/51250378](https://stackoverflow.com/a/51250378)
